src/dataset/domain/useCases/createDataset.ts

```diff
@@ -0,0 +1,47 @@
+export interface CreateDatasetFormFields {
+  createDatasetTitle: string
+}
+
+export interface FormValidationResult {
+  isValid: boolean
+  errors: Record<keyof CreateDatasetFormFields, string | undefined>
+}
+
+export interface FormValidationService {
+  validateForm: (fields: CreateDatasetFormFields) => FormValidationResult
+}
+
+export const formValidationService: FormValidationService = {
```

Comment:
Since we're not using the Service suffix in the rest of use cases of the application, we should remove this suffix from these use cases

Instead you can give the use case a name to explain the use case such as `formValidationService` =>`validateDataset()` and `formSubmissionService` => `createDataset()`

Reply:
Can you verify the new names are more appropriate?